### ISSUE TITLE
getNotificationAppLaunchDetails on iOS returning false without payload

### ISSUE THREAD

**Issue by d3xt3r2909** [MaikuB/flutter_local_notifications#512]
Hi @MaikuB, thank you for the great library. 

I have an issue where `getNotificationAppLaunchDetails` is returning me always false and payload null as a result on the **iOS** platform, while on Android these values are good, as should be. 

**To Reproduce**
If applicable, steps to reproduce the behavior:
1. Downloaded your example project
2. Added part of code to function `onSelectNotification` callback (like on image in the attachment), just to access and print payload and got the result with false & null
3. When you do the same thing on the android platform, you will have payload and true as value for `didNotificationLaunchApp`

![image](https://user-images.githubusercontent.com/17637299/75695274-1bdebf80-5caa-11ea-960f-74560400e35e.png)

Goal: I would love to be the same results on both platforms if it is possible.

**Comment by MaikuB** [comment]
Sorry but the steps you described aren't sufficient to show how to reproduce the problem. I'm using the example app and don't see the issue you've described. `didNotificationLaunchApp` only returns true if the app was killed and you tapped on a notification (that was created by the app) to launch the app. None of your steps actually describe what you did with the notification.

**Comment by d3xt3r2909** [comment]
Hi @MaikuB, sorry if my first post is not clear enough for you.

Can you please open your example project and replace code from line 61 to 67 :

```
await flutterLocalNotificationsPlugin.initialize(initializationSettings,
      onSelectNotification: (String payload) async {
    if (payload != null) {
      debugPrint('notification payload: ' + payload);
    }
    selectNotificationSubject.add(payload);
  });
```

with this code: 

```
await flutterLocalNotificationsPlugin.initialize(initializationSettings,
      onSelectNotification: (String payload) async {
    if (payload != null) {
      debugPrint('notification payload: ' + payload);
    }

    final NotificationAppLaunchDetails details =
        await flutterLocalNotificationsPlugin.getNotificationAppLaunchDetails();

    print("test me !-> ${details.payload}");

    selectNotificationSubject.add(payload);
  });
```

Then, 

1. Click on the first button "Show plain notification with payload" 
2. Tap on the notification that has shown up
3. Of course `onSelectNotification` will be triggered

Now, on the Android side result of line `print("test me !-> ${details.payload}");` will be "item x", while on same scenario, on iOS platform, this line of code will print null. 

> didNotificationLaunchApp only returns true if the app was killed and you tapped on a notification (that was created by the app) to launch the app.

This basically, resolves my question, because I was using another way for handling tap on the notification, and fetching data of click, based on `getNotificationAppLaunchDetails();`. Now, the only question is, why the behavior for this function is different for Android and iOS? 

Also, is it possible to make a function that will return your payload of last shown notification (notification that is on the stack, and that user didn't already click on it?)

It would be very nice if you can add also an example of the usage of `getNotificationAppLaunchDetails`  to your example project.

**Comment by MaikuB** [comment]
Thanks for the info. Was able to reproduce but will need to see if it's possible to fix. The behaviour is different because the mechanisms on each platform are different.

Regarding getting notifications that are in still in the notifications centre/drawer (i.e. the active notifications), It's not possible via this plugin but there are native APIs that can get those notifications. They're supported by the plugin as these APIs can't be used by all supported versions of Android and iOS. They're only available for Android 6.0+ and iOS 10+. If your app will only be supporting those at a minimum then you'll need to write your own code to use those APIs within your app.

**Comment by d3xt3r2909** [comment]
> Regarding getting notifications that are in still in the notifications centre/drawer (i.e. the active notifications), It's not possible via this plugin but there are native APIs that can get those notifications. They're supported by the plugin as these APIs can't be used by all supported versions of Android and iOS. They're only available for Android 6.0+ and iOS 10+. If your app will only be supporting those at a minimum then you'll need to write your own code to use those APIs within your app.

Ok, understand. I will use the listener, or help of dartrx to listen events of opening notification on some other place in my code. 

> Thanks for the info. Was able to reproduce but will need to see if it's possible to fix. The behaviour is different because the mechanisms on each platform are different.

Yep, it would be cool, or at least, you can document this somewhere in readme.

Thanks for quick response,

All the best

**Comment by MadhanHoggy** [comment]
Getting the same issue in Android version :(
I have upgraded the flutter_local_notification package to 9.7.0 , but still

**Comment by nickcuypers** [comment]
@MadhanHoggy try this:

```
final NotificationAppLaunchDetails details =
        await flutterLocalNotificationsPlugin.getNotificationAppLaunchDetails();

    print("test me !-> ${details.notificationResponse.payload}");
```

**Comment by tal412** [comment]
> Getting the same issue in Android version :( I have upgraded the flutter_local_notification package to 9.7.0 , but still `details.payload` displays null
> 
> ```
>     final NotificationAppLaunchDetails details =
>         await flutterLocalNotificationsPlugin.getNotificationAppLaunchDetails();
> 
>     print("test me !-> ${details.payload}");
> ```

Having the same issue on the latest iOS version...

**Comment by tal412** [comment]
Having this issue on the latest iOS version.
No matter where I put it, `didNotificationLaunchApp` is always false.

**Comment by Fxynos** [comment]
Hi! I had the same issue on iOS for `18.0.1` with `getNotificationAppLaunchDetails`: `didNotificationLaunchApp` returning `false` and null `payload`. Tried older package versions, didn't help.
I fixed it by registering notification center delegate in `ios/Runner/AppDelegate.swift` like this:
```swift
// ...
import UserNotifications

@main
@objc class AppDelegate: FlutterAppDelegate {
  override func application(
    _ application: UIApplication,
    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
  ) -> Bool {
  // ...
  if #available(iOS 10.0, *) {
      UNUserNotificationCenter.current().delegate = self as? UNUserNotificationCenterDelegate
  }
  // ...
  }
}
```
